test-framework/sudo-compliance-tests/src/lecture_file.rs

```diff
@@ -0,0 +1,97 @@
+use sudo_test::{Command, Env};
+use crate::{Result, SUDOERS_ROOT_ALL, USERNAME, SUDOERS_NEW_LECTURE, SUDOERS_NEW_LECTURE_USER};
+
+#[test]
+fn default_lecture_message() -> Result<()> {
+    let expected = format!(
+        "\nWe trust you have received the usual lecture from the local System\nAdministrator. It usually boils down to these three things:\n\n    #1) Respect the privacy of others.\n    #2) Think before you type.\n    #3) With great power comes great responsibility."
+    );
+    let env = Env(SUDOERS_ROOT_ALL).user(USERNAME).build()?;
+
+    let output = Command::new("sudo")
+    .arg(format!("ls"))
+    .as_user(USERNAME)
+    .exec(&env)?;
+    assert_eq!(Some(1), output.status().code());
+    assert_eq!(false, output.status().success());
+    if sudo_test::is_original_sudo() {
+        assert_contains!(
+            output.stderr(),
+            expected
+        );
+    }
+    Ok(())
+}
+
+#[test]
+#[ignore]
+fn new_lecture_message() -> Result<()> {
+    let new_lecture = format!("I <3 sudo");
+    let env = Env([
+        SUDOERS_ROOT_ALL,
+        SUDOERS_NEW_LECTURE])
+        .file("/etc/sudo_lecture", new_lecture)
```

Comment:
another interesting scenario to test is: what happens if `/etc/sudo_lecture` is world writable? ogsudo emits an error when `/etc/sudoers` is world-writable. is that also the case for `/etc/sudo_lecture`? (I do not know the answer)

Reply:
those should also be tests: "no lecture is shown if lecture file is XYZ". you should test these scenarios: https://github.com/memorysafety/sudo-rs/blob/a6942a5e88177cd6c0f055375e7af014f9f0d287/test-framework/sudo-compliance-tests/src/sudoers.rs#L13-L61